### PR TITLE
CP-23647: Add SRIOV VF IP collection

### PR DESCRIPTION
**Pull request by xiewei20082008**
Signed-off-by: Wei Xie <wei.xie@citrix.com>

### CODE REVIEW

**Comment by robertbreker** [comment]
CI failed - Please run "gofmt -s -w ." before committing.

File: guestmetric/guestmetric_linux.go [hunk 1]
```diff
@@ -149,11 +149,31 @@ func (c *Collector) CollectNetworkAddr() (GuestMetric, error) {
 		}
 		paths = append(paths, prefixPaths...)
 	}
-
 	for _, path := range paths {
 		iface := filepath.Base(path)
 		vifId, err := getVifId(iface)
 		if err != nil {
+			macAddress, err := readSysfs(path+"/address")
```
Comment (robertbreker) [inline guestmetric/guestmetric_linux.go]: Can we please move this new section into a separate function?
Reasoning:
- It's not obvious to me what this code is there for. Needs a descriptive function name,
- It's nested too deep. That makes it hard to read.

Comment (xiewei20082008) [reply guestmetric/guestmetric_linux.go]: Sure. Will tweak it later on. Thank you very much.

**Comment by cheng-z** [comment]
Thanks @robertbreker  for the review.
Please help to refine the code as suggest by @robertbreker and I will have a look again and merge if no other concern.
Thanks.

**Comment by cheng-z** [comment]
Hi @robertbreker  Could you please help to have a look at the code again?
Thanks in advance.

**Comment by cheng-z** [comment]
Thank you all for the review. Merge

**Pull request merged**